File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/domain/splash/Splash.java
package com.crpr.androidcinema.domain.splash;

import android.app.Activity;

import com.crpr.androidcinema.domain.common.Result;
import com.crpr.androidcinema.domain.welcome_wizard.WelcomeWizardResult;
import com.crpr.androidcinema.presentation.common.Base;

import rx.Observable;

/**
 * Created by claudioribeiro on 09/07/16.
 */
public interface Splash {

    interface View extends Base.View {
        void goToNextActivity(Class<?> klass);
    }

    interface Presenter extends Base.Presenter {
        void getConfiguration();
        void onReceiveResult(WelcomeWizardResult result);
    }

    interface Interactor {
        Observable<WelcomeWizardResult> start();
    }

    interface Navigator {
        void navigate(Activity activity, Class<?> klass);
    }
}

File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/domain/splash/Splash.kt
package com.crpr.androidcinema.domain.splash

import android.app.Activity

import com.crpr.androidcinema.domain.common.Result
import com.crpr.androidcinema.domain.welcome_wizard.WelcomeWizardResult
import com.crpr.androidcinema.presentation.common.Base

import rx.Observable

/**
 * Created by claudioribeiro on 09/07/16.
 */
interface Splash {

    interface View : Base.View {
        fun goToNextActivity(klass: Class<*>)
    }

    interface Presenter : Base.Presenter {
        fun getConfiguration()
        fun onReceiveResult(result: WelcomeWizardResult)
    }

    interface Interactor {
        fun start(): Observable<WelcomeWizardResult>
    }

    interface Navigator {
        fun navigate(activity: Activity, klass: Class<*>)
    }
}
